### PR TITLE
multiple field value added

### PR DESCRIPTION
**Pull request by cs-pranali-chavan**
multiple field value rendering done in view controller

Screenshot 2024-12-13 at 4 43 07 PM

Screenshot 2024-12-13 at 4 49 42 PM

### CODE REVIEW

File: widget/view.controller.js [hunk 1]
```diff
@@ -4,76 +4,104 @@
...
+    function init() {
+      // To handle backward compatibility for widget
+      _handleTranslations();
+      checkCurrentPage($scope.pageState);
+      $scope.picklistField =  $scope.config.picklistField;
+      $scope.multipleFieldsItems= $scope.config.multipleFieldsItems || [];
+      var moduleMetaData = modelMetadatasService.getMetadataByModuleType($scope.config.resource);
+      //to check if dataSource is present and fetch data from connector action or else from API query
+      if(moduleMetaData.dataSource){ 
+        var _promises = [executeConnectorAction(moduleMetaData.dataSource)];
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:37]: what is a diff between executeConnectorAction and executeConnectorOperation?

File: widget/view.controller.js [hunk 2]
```diff
@@ -4,76 +4,104 @@
...
-          let params = $scope.pageState.current.params;
-          $scope.indicator = params.id;
+    function init() {
+      // To handle backward compatibility for widget
+      _handleTranslations();
+      checkCurrentPage($scope.pageState);
+      $scope.picklistField =  $scope.config.picklistField;
+      $scope.multipleFieldsItems= $scope.config.multipleFieldsItems || [];
+      var moduleMetaData = modelMetadatasService.getMetadataByModuleType($scope.config.resource);
+      //to check if dataSource is present and fetch data from connector action or else from API query
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:35]: You can add more details  about why data is fetched from the connector

File: widget/view.controller.js [hunk 3]
```diff
@@ -4,76 +4,104 @@
...
+      let _connectorName = _moduleMetaData.connector;
+      let _connectorAction = _moduleMetaData.actions.risk_distribution;
+      let payload = { 'indicator': $scope.indicator || '8.8.8.8' };
+      return heatmapService.executeAction(_connectorName, _connectorAction, payload)
+    }
+
+    function executeConnectorOperation(_moduleMetaData){
+      let _connectorName = _moduleMetaData.connector;
+      let _connectorAction = _moduleMetaData.operation;
+      let payload = { 'indicator': $scope.indicator || '8.8.8.8' };
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js]: why 8.8.8.8?

Comment (cs-suyog-jinturkar) [reply widget/view.controller.js]: can we just bailout if there is no ioc

File: widget/view.controller.js [hunk 4]
```diff
@@ -4,76 +4,104 @@
...
+    }
+
+    function setDistributionData(_responseData){
+      $scope.distribution = angular.copy(_responseData);
+      $scope.distributionSegment = 100 / _responseData.length; //width of each section
+      $scope.processing = false;
+    }
 
-      init();
+    function setMultipleFieldsData(_operationData){
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js]: can we call this a titleFieldsData? or summaryFieldsData?

File: widget/view.controller.js [hunk 5]
```diff
@@ -4,76 +4,104 @@
...
-    function heatmap100Ctrl($scope, widgetUtilityService, heatmapService, config, modelMetadatasService, $state) {
-      $scope.config = config;
-      $scope.processing = true;
-      $scope.pageState = $state;
+  function heatmap100Ctrl($scope, widgetUtilityService, heatmapService, config, modelMetadatasService, $state, $q) {
+    $scope.config = config;
+    $scope.processing = true;
+    $scope.pageState = $state;
+    $scope.multipleFieldsItemsData = [];
+    $scope.connectorOperationResponse = {};
```
Comment (cs-suyog-jinturkar) [inline widget/view.controller.js:18]: connectorOperationResponse --> connectorResponse

File: widget/view.controller.js (hunk 1 above)
Comment (cs-pranali-chavan) [reply widget/view.controller.js:37]: executeConnectorAction - to call connector action from data source 
executeConnectorOperation - to call connector operation from data source

File: widget/view.controller.js (hunk 3 above)
Comment (cs-pranali-chavan) [reply widget/view.controller.js]: yes removing it was for testing purpose

File: widget/view.controller.js (hunk 4 above)
Comment (cs-pranali-chavan) [reply widget/view.controller.js]: update to titleFieldsData

File: widget/view.controller.js (hunk 5 above)
Comment (cs-pranali-chavan) [reply widget/view.controller.js:18]: connectorOperationResponse keeping same as operation /action response

**Pull request merged**